libraries/ghc-prim/cbits/pdep.c

```diff
@@ -0,0 +1,71 @@
+#include "Rts.h"
+#include "MachDeps.h"
+
+extern StgWord hs_pdep8(StgWord src);
+StgWord
+hs_pdep8(StgWord src, StgWord mask)
+{
+  return hs_pdep64(src, mask);
+}
+
+extern StgWord hs_pdep16(StgWord src);
+StgWord
+hs_pdep16(StgWord src, StgWord mask)
+{
+  return hs_pdep64(src, mask);
+}
+
+extern StgWord hs_pdep32(StgWord src);
+StgWord
+hs_pdep32(StgWord src, StgWord mask)
+{
+  return hs_pdep64(src, mask);
+}
+
+extern StgWord hs_pdep64(StgWord64 src);
+StgWord
+hs_pdep64(StgWord src, StgWord mask)
```

Comment:
The declaration signatures don't match the definition signatures. Also, doesn't `hs_pdep64` need to be forward-declared since it's used by the others?